RIOT-OS/Release-Specs
# testutils: Automated testing environment for release tests using pytest

### Contribution description

Third time's a charm: after https://github.com/RIOT-OS/Release-Specs/pull/79 and some major overhaul of that framework to use (then) `riotnode` (now `riotctrl`) and `pytest` by @fjmolinas, I ported the latter to run with the new [`riotctrl`](https://github.com/RIOT-OS/riotctrl/) python package and the in-tree [`riotctrl_shell`](https://github.com/RIOT-OS/RIOT/tree/b05b2fddb967b170f2c2cdcfda5641085d55fbcc/dist/pythonlibs/riotctrl_shell) module.

I also cleaned up the by now very complicated history of that effort and squashed a lot of commits. Since the most of the porting work was done by @fjmolinas (I just moved some code around and enhanced his code) I made him the main author of the work in the commit history, but added all co-authors to the respective commits for the changes they made.

### Testing procedure
Run `tox` and the tests will be run (there are none at the moment, see #160, #161, #162, #163, #158, and #159).
Check-out #164 to run all tests.
There is also some more info in the README how to select specific groups of tests.

Self-tests are run by Github Actions and can be run locally with `tox -- --self-test`.

### Issues/PRs references
Alternative to #79.

Required for #160, #161, #162, #163, #158, and #159
